staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest.go

```diff
@@ -443,7 +445,19 @@ func createHandler(r rest.NamedCreater, scope RequestScope, typer runtime.Object
 		}
 		trace.Step("Self-link added")
 
-		responsewriters.WriteObject(ctx, http.StatusCreated, scope.Kind.GroupVersion(), scope.Serializer, result, w, req)
+		// If the object is partially initialized, always indicate it via StatusAccepted
+		code := http.StatusCreated
+		if accessor, err := meta.Accessor(result); err == nil {
+			if accessor.GetInitializers() != nil {
+				code = http.StatusAccepted
```

Comment:
I think in isolation (just this method) this block of code is correct.  If the storage decides to return an object from CreateInitialized, then the result was accepted, not created.  We don't have that case today, but I don't think the "wait forever" semantic is a detail this code should know about.